### PR TITLE
Improve the default fallback handler

### LINKED ISSUE #57: How to figure out the the URL when 'No matching mock handler' exception occurs?
I'm getting a `No matching mock handler` in my unit test which is most likely caused by a typo in my expectation:

```csharp
var mockHttp = new MockHttpMessageHandler();
mockHttp.Expect(HttpMethod.Get, "...the url...").Respond("application/json", "...the response...");
```

but I can't figure out what I typed wrong. Is there any way to get the actual URL?

### PR DESCRIPTION
**Pull request by Jericho**
Include the HTTP verb and the URL in the exception if no matching mock handler is found

Resolves #57

### CODE REVIEW

**Comment by Jericho** [comment]
FYI: the FallbackResponse property in `MockHttpMessageHandler.cs` has been marked as obsolete since December 2014. Do you mind if I remove it?

File: RichardSzalay.MockHttp.Shared/MockHttpMessageHandler.cs [hunk 1]
```diff
@@ -214,13 +217,6 @@ public HttpResponseMessage FallbackResponse
             }
         }
 
-        HttpResponseMessage CreateDefaultFallbackMessage()
```
[the remark this answers is not in the document]
Comment (Jericho) [reply RichardSzalay.MockHttp.Shared/MockHttpMessageHandler.cs]: Done

File: RichardSzalay.MockHttp.Tests/MockHttpMessageHandlerTests.cs [hunk 2]
```diff
@@ -157,26 +157,6 @@ public void Should_return_fallback_for_unmatched_requests()
             Assert.Equal("Awesome", result.ReasonPhrase);
         }
 
-        [Fact]
-        public void Should_return_fallbackresponse_for_unmatched_requests()
```
Comment (richardszalay) [inline RichardSzalay.MockHttp.Tests/MockHttpMessageHandlerTests.cs]: Why did you delete these tests?

Comment (Jericho) [reply RichardSzalay.MockHttp.Tests/MockHttpMessageHandlerTests.cs]: Because they are unit testing the FallbackResponse property which we just deleted.

Comment (Jericho) [reply RichardSzalay.MockHttp.Tests/MockHttpMessageHandlerTests.cs]: Also, the behavior that was being unit tested in `Should_return_fallbackresponse_for_unmatched_requests` against the `FallbackResponse` property is unit tested in `Should_return_fallback_for_unmatched_requests` against the `Fallback` property and the `Should_not_match_expect_out_of_order`  unit test relied on `FallbackResponse` having a property named `StatusCode`. Fallback does not have have such a property so I couldn't figure out a obvious replacement for the unit test.

Comment (richardszalay) [reply RichardSzalay.MockHttp.Tests/MockHttpMessageHandlerTests.cs]: Gotcha. I agree with the first, but `Should_not_match_expect_out_of_order` is a critical piece of behavior so I'll take your branch and reimplement it before merging.

**Comment by richardszalay** [comment]
The commits have been squashed and released as 4.0.0
